### DISCUSSION TITLE
Is it possible to use a trace tool for educational purposes?

### DISCUSSION THREAD

**Discussion by lcgamboa** [feilipu/Arduino_FreeRTOS_Library#140] [Q&A]
I did some tests with [FreeRTOS-barectf](https://github.com/kuopinghsu/FreeRTOS-barectf) but I had little success. I used a mega2560, which only partially worked when compiled without optimization (-O0).
I use this excellent library for my practical classes on real-time systems, a trace tool would make it easier to explain to students.
Using this version of FreeRTOS for the AVR is a great option for teaching, as it compiles much faster than an ESP32 or STM32.

**Comment by feilipu** [comment]
It is possible to compile programs utilising FreeRTOS for ATmega using the [Microchip Studio IDE](https://onlinedocs.microchip.com/oxy/GUID-ECD8A826-B1DA-44FC-BE0B-5A53418A47BD-en-US-12/index.html), which has an integrated trace and debug tool set. But this is of course quite a big step away from the Arduino IDE, increasing the complexity of teaching substantially.

I don't think there's anything integrated with Arduino that would help out too much, unfortunately.
